src/itn_uptime_analyzer/uptime.go

```diff
@@ -82,56 +81,65 @@ func (identity Identity) GetUptime(config AppConfig, sheet *sheets.Service, ctx
 
 					if submissionDataToday.GraphqlControlPort != 0 {
 						if (identity.publicKey == submissionDataToday.Submitter.String()) && (identity.publicIp == submissionDataToday.RemoteAddr) && (*identity.graphQLPort == strconv.Itoa(submissionDataToday.GraphqlControlPort)) {
-							if lastSubmissionTimeString != "" {
-								lastSubmissionTime, err = time.Parse(time.RFC3339, lastSubmissionTimeString)
-								if err != nil {
-									log.Fatalf("Error parsing time: %v\n", err)
-								}
-							}
 
 							currentSubmissionTime, err := time.Parse(time.RFC3339, submissionDataToday.CreatedAt)
 							if err != nil {
 								log.Fatalf("Error parsing time: %v\n", err)
 							}
 
-							if (lastSubmissionTimeString != "") && (currentSubmissionTime.After(lastSubmissionTime.Add(time.Duration(syncPeriod-5) * time.Minute))) && (currentSubmissionTime.Before(lastSubmissionTime.Add(time.Duration(syncPeriod+5) * time.Minute))) {
-								uptimeYesterday = append(uptimeYesterday, true)
+							if lastSubmissionTimeString != "" {
+								lastSubmissionTime, err = time.Parse(time.RFC3339, lastSubmissionTimeString)
+								if err != nil {
+									log.Fatalf("Error parsing time: %v\n", err)
+								}
+							} else {
+								uptimeToday = append(uptimeToday, true)
 								lastSubmissionTimeString = submissionDataToday.CreatedAt
-							} else if lastSubmissionTimeString == "" {
-								uptimeYesterday = append(uptimeYesterday, true)
+								continue
+							}
+
+							if (lastSubmissionTimeString != "") && (currentSubmissionTime.After(lastSubmissionTime.Add(time.Duration(syncPeriod-5) * time.Minute))) {
+								uptimeToday = append(uptimeToday, true)
 								lastSubmissionTimeString = submissionDataToday.CreatedAt
+								continue
+							} else {
+								continue
 							}
```

Comment:
```suggestion
							}
							continue
```

Reply:
I feel that the overall logic in this function can be expressed more clearly.

Reply:
We can talk about this